From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jollamobile.com>
Date: Mon, 4 May 2015 17:20:52 +0300
Subject: [PATCH] Remove optimization not to follow keyboard state if not
 present

Context property includes both internal and external hardware
keyboard state. Thus, if keyboard wasn't connected during startup,
its state wasn't tracked.
---
 src/mimhwkeyboardtracker.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/mimhwkeyboardtracker.cpp b/src/mimhwkeyboardtracker.cpp
index a94f35f..471db3f 100644
--- a/src/mimhwkeyboardtracker.cpp
+++ b/src/mimhwkeyboardtracker.cpp
@@ -45,11 +45,9 @@ MImHwKeyboardTrackerPrivate::MImHwKeyboardTrackerPrivate(MImHwKeyboardTracker *q
     present(false)
 {
 #ifdef HAVE_CONTEXTSUBSCRIBER
-    ContextProperty keyboardPresentProperty(keyboardPresent);
     keyboardOpenProperty.reset(new ContextProperty(keyboardOpen));
-    keyboardPresentProperty.waitForSubscription(true);
     keyboardOpenProperty->waitForSubscription(true);
-    present = keyboardPresentProperty.value().toBool();
+    present = true; // assume keyboard present as context property currently includes also external keyboards
     if (present) {
         QObject::connect(keyboardOpenProperty.data(), SIGNAL(valueChanged()),
                          q_ptr, SIGNAL(stateChanged()));
-- 
2.35.1

